Automattic/simplenote-ios
# Nü History Slider

### Details:
In this PR we're implementing a brand new History Slider. This is @eshurakov 's work, opening the PR in order to give it a bit of more visibility.

Depends on #803 (and further testing!).

Closes #762

### Test
1. Open a note
2. Open note options
3. Open note history

- [ ] If no internet connection, inline error message is shown
- [x] If current version is selected, restore button is disabled
- [x] On devices with haptic feedback, dragging a slider triggers haptic feedback
- [x] Screen can be rotated to landscape / portrait
- [x] Selecting a version updates note editor with that version (with alpha transition animation)
- [x] Selecting a version and dismissing a history slider restores editor to the current version
- [x] Selecting a version and saving it should persist the changes
- [x] Dismiss the card by swiping down
- [x] Dismiss the card by pressing "X"
- [x] Pull the card a little down and back up. It shouldn't dismiss the card.
- [x] Check that overlay blocks interaction with the editor
- [x] Tap outside of the card to dismiss
- [x] Tap inside of the card, view should not get dismissed
- [x] Check that all elements are accessible with VoiceOver

### Test (from search)
1. Search notes
2. Open one of the note

- [x] Check that opening history clears text highlights in the editor

### Test (slow network)
1. Setup slow network
2. Open a note
3. Open note options
4. Open note history

- [x] As you pick different versions, observe progress indicator as versions are being loaded
- [x] Select one of the versions that wasn't loaded yet. Wait till the version is loaded and check that UI of Version Slider is updated (date, button) and editor content is updated to reflect the selected version.



### Release
`RELEASE-NOTES.txt` was updated in 36046a3 with:
> New Note History interface #762
